### PR TITLE
Validate trim aoi

### PR DESCRIPTION
**Pull request by feenster**

### CODE REVIEW

File: server/api/grid_apis.py [hunk 1]
```diff
@@ -72,6 +73,8 @@ def put(self):
         try:
             grid = GridService.trim_grid_to_aoi(grid_dto)
             return grid, 200
+        except (InvalidGeoJson) as e:
```
Comment (hunt3ri) [inline server/api/grid_apis.py]: No brackets need around the exception

File: server/api/grid_apis.py [hunk 2]
```diff
@@ -72,6 +73,8 @@ def put(self):
...
+            return {"error": f'{str(e)}'},
```
Comment (hunt3ri) [inline server/api/grid_apis.py]: You need to return a 400 status here.

File: server/services/grid_service.py [hunk 3]
```diff
@@ -32,6 +33,15 @@ def trim_grid_to_aoi(grid_dto: GridDTO) -> geojson.FeatureCollection:
 
         # create a shapely shape from the aoi
         aoi_multi_polygon_geojson = GridService.merge_to_multi_polygon(aoi, dissolve=True)
+
```
Comment (hunt3ri) [inline server/services/grid_service.py]: As discussed, testing we have a multipolygon should be responsibility of the merge_to_multi_polygon function

File: server/api/grid_apis.py (hunk 1 above)
Comment (feenster) [inline server/api/grid_apis.py]: done

File: server/api/grid_apis.py (hunk 2 above)
Comment (feenster) [inline server/api/grid_apis.py]: done

File: server/services/grid_service.py (hunk 3 above)
Comment (feenster) [inline server/services/grid_service.py]: done

**Pull request merged**